frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideInBranchBelowAction.java

```diff
@@ -230,27 +277,58 @@ String createOrCheckoutNewBranch(Project project, GitRepository gitRepository, S
         initialName,
         /* showCheckOutOption */ true,
         /* showResetOption */ true,
-        /* showSetTrackingOption */ true);
+        /* showSetTrackingOption */ false); // TODO (#496): use setTrackingOption that comes with 2020.2
 
     GitNewBranchOptions options = gitNewBranchDialog.showAndGetOptions();
 
     if (options == null) {
       log().debug("Name of branch to slide in is null: " +
           "most likely the action has been canceled from create-new-branch dialog");
-      return null;
+      return Tuple.of(null, () -> {});
     }
 
     var branchName = options.getName();
     if (!initialName.equals(branchName)) {
-      return branchName;
+      return Tuple.of(branchName, () -> {});
     }
 
-    if (options.shouldCheckout()) {
-      checkoutOrReset(project, repositories, startPoint, options);
-    } else {
-      createNewBranch(project, repositories, startPoint, options);
+    Runnable preSlideInRunnable = () -> {};
+    var remoteBranch = getGitRemoteBranch(project, gitRepository, branchName);
+
+    if (options.shouldCheckout() && remoteBranch != null) {
+      preSlideInRunnable = () -> checkoutRemoteBranch(project, repositories, remoteBranch.getName());
+
+    } else if (!options.shouldCheckout() && remoteBranch != null) {
+      var refspec = "refs/remotes/${remoteBranch.getName()}:refs/heads/${branchName}";
+      preSlideInRunnable = () -> new FetchBackgroundable(project, gitRepository, GitRemote.DOT.getName(), refspec,
+          "Fetching Remote Branch").queue();
+
+    } else if (options.shouldCheckout() && remoteBranch == null) {
+      preSlideInRunnable = () -> checkoutOrReset(project, repositories, startPoint, options);
+
+    } else if (!options.shouldCheckout() && remoteBranch == null) {
+      preSlideInRunnable = () -> createNewBranch(project, repositories, startPoint, options);
     }
 
-    return options.getName();
+    return Tuple.of(options.getName(), preSlideInRunnable);
+  }
+
+  @Nullable
+  private static GitRemoteBranch getGitRemoteBranch(Project project, GitRepository gitRepository, String branchName) {
+    var remotes = gitRepository.getRemotes();
+    if (remotes.size() > 0) {
+      var remote = Option.ofOptional(remotes.stream().findFirst()).getOrNull();
+      assert remote != null : "remote is null";
+
+      if (remotes.size() > 1) {
+        VcsNotifier.getInstance(project).notifyInfo(format(
+            getString("action.GitMachete.BaseSlideInBranchBelowAction.notification.message.multiple-remotes"),
+            remote.getName()));
+      }
+
+      var remoteBranchName = "${remote.getName()}/${branchName}";
+      return gitRepository.getBranches().findRemoteBranch(remoteBranchName);
```

Comment:
Hmm I don't think that's the most convenient logic for the user: I'd rather only take into account the remotes that have a matching branch, and from them pick the first one in case there's more than one

Reply:
(or actually, from the remote repos that have a matching branch, I'd prefer `origin` first if applicable)